dataactbroker/handlers/fileHandler.py

```diff
@@ -266,15 +267,34 @@ def upload(file_ref, file_type, app, current_user):
                 t.start()
                 t.join()
             api_response = {"success": "true", "submission_id": submission.submission_id}
-            return JsonResponse.create(StatusCode.OK, api_response)
+            json_response = JsonResponse.create(StatusCode.OK, api_response)
         except (ValueError, TypeError, NotImplementedError) as e:
-            return JsonResponse.error(e, StatusCode.CLIENT_ERROR)
+            json_response = JsonResponse.error(e, StatusCode.CLIENT_ERROR)
         except ResponseException as e:
             # call error route directly, status code depends on exception
-            return JsonResponse.error(e, e.status)
+            json_response = JsonResponse.error(e, e.status)
         except Exception as e:
-            # unexpected exception, this is a 500 server error
-            return JsonResponse.error(e, StatusCode.INTERNAL_ERROR)
+            # handle unexpected exception as a 500 server error
+            json_response = JsonResponse.error(e, StatusCode.INTERNAL_ERROR)
+        finally:
+            # handle a missing JSON response
+            if json_response is None:
+                json_response = JsonResponse.error(Exception("Failed to catch exception"), StatusCode.INTERNAL_ERROR)
+
+            # handle errors within upload jobs
+            if json_response.status_code != StatusCode.OK and submission:
+                jobs = sess.query(Job).filter(Job.submission_id == submission.submission_id,
+                                              Job.job_type_id == JOB_TYPE_DICT['file_upload'],
+                                              Job.job_status_id == JOB_STATUS_DICT['running'],
+                                              Job.file_type_id.in_([FILE_TYPE_DICT_LETTER_ID['A'],
```

Comment:
Do we always want to set all 3 to failed? What if we're only reuploading one file? We don't want to fail the other 3 jobs do we?

Reply:
This should only set `running` Jobs to `failed`.

Reply:
I can read... I swear